sql/core/src/test/scala/org/apache/spark/sql/execution/command/ShowTablesSuiteBase.scala

```diff
@@ -40,6 +42,10 @@ trait ShowTablesSuiteBase extends QueryTest with DDLCommandTestUtils {
     checkAnswer(df, expected)
   }
 
+  after {
+    spark.sessionState.catalogManager.reset()
```

Comment:
@cloud-fan I need to add this because the current namespace is not reset even if the dropped namespace is the current namespace. (this introduced a bug in test, which I fixed below).

If I have:
```
sql("create database db")
sql("create table spark_catalog.db ...")
sql("use spark_catalog.db") // Current namespace is "db"
sql("drop database db") // Current namespace is still "db"
```
Should we fix above? I can fix it as a separate PR if the current behavior doesn't seem right.

Reply:
what's the behavior today (with v1 commands)?

Reply:
V1 / V2 behaviors are the same since the current namespace is managed by `CatalogManager`. (`SessionCatalog` doesn't reset the current database when the current database is dropped). Let me check the postgre's behavior.

Reply:
`postgres` doesn't allow to drop the database you are in: `[23877] ERROR:  cannot drop the currently open database`.

`mysql` resets the current database to `NULL`:
```
CREATE DATABASE tmp;
USE tmp;
select database();
drop database tmp;
select database();

database()
tmp
database()
NULL
```

Reply:
I think this problem is inevitable. It's possible that another user drops the database you are using currently via another session, and there is no way to forbid it.

It seems more of a test problem: the test should reset any stats it set during testing, including the current database.